servers/src/main/java/tachyon/master/next/MasterInfo.java

```diff
@@ -0,0 +1,294 @@
+import tachyon.thrift.DependencyDoesNotExistException;
+import tachyon.thrift.DependencyInfo;
+import tachyon.thrift.FileBlockInfo;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.FileInfo;
+import tachyon.thrift.InvalidPathException;
+import tachyon.thrift.WorkerInfo;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.underfs.UnderFileSystem.SpaceType;
+
+/**
+ * A wrapper class of BlockMaster, FileSystemMaster and other master modules to provide a view of
+ * their containing information.
+ */
+public final class MasterInfo {
+  private final BlockMaster mBlockMaster;
+  private final FileSystemMaster mFileSystemMaster;
+  private final InetSocketAddress mMasterAddress;
+  private final long mStartTimeMs;
+  private final TachyonConf mTachyonConf;
+  private final String mUFSDataFolder;
+
+  public MasterInfo(BlockMaster blockMaster, FileSystemMaster fileSystemMaster,
+      InetSocketAddress address, TachyonConf tachyonConf) {
+    mBlockMaster = Preconditions.checkNotNull(blockMaster);
+    mFileSystemMaster = Preconditions.checkNotNull(fileSystemMaster);
+    mMasterAddress = Preconditions.checkNotNull(address);
+    mTachyonConf = Preconditions.checkNotNull(tachyonConf);
+    mUFSDataFolder = mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER, Constants.DEFAULT_DATA_FOLDER);
+    mStartTimeMs = System.currentTimeMillis();
+  }
+
+  /**
+   * Gets the capacity of the whole system.
+   *
+   * @return the system's capacity in bytes.
+   */
+  public long getCapacityBytes() {
+    return mBlockMaster.getCapacityBytes();
+  }
+
+  /**
+   * Gets the list of block info of an InodeFile determined by path.
+   *
+   * @param path path to the file
+   * @return The list of the block info of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<BlockInfo> getBlockInfoList(TachyonURI path)
+      throws InvalidPathException, FileDoesNotExistException {
+    return mFileSystemMaster.getBlockInfoList(path);
+  }
+
+  /**
+   * Gets the list of blocks of an {@link InodeFile} determined by path.
+   *
+   * @param path path to the file
+   * @return The list of the blocks of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<FileBlockInfo> getFileBlockList(TachyonURI path)
+      throws InvalidPathException, FileDoesNotExistException {
+    long fileId = mFileSystemMaster.getFileId(path);
+    return getFileBlockList(fileId);
+  }
+
+  /**
+   * Gets the list of blocks of an {@link InodeFile} determined by the file id.
+   *
+   * @param fileId id of the file
+   * @return The list of the blocks of the file
+   * @throws InvalidPathException when the path is invalid
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public List<FileBlockInfo> getFileBlockList(long fileId)
```

Comment:
Good call. I added them for consistency. The web ui uses fileId, but I found some other methods use path, e.g. getBlockInfoList, so I added both. What do you think? I can retain only the necessary ones, and add back others when needed.